js/researches/german/GermanParticiple.js

```diff
@@ -0,0 +1,83 @@
+var exceptionsRegex =
+	/\S+(apparat|arbeit|dienst|haft|halt|keit|kraft|not|pflicht|schaft|schrift|tät|wert|zeit)($|[ \n\r\t\.,'\(\)\"\+\-;!?:\/»«‹›<>])/ig;
+var exceptionsParticiplesActive = require( "./passivevoice-german/exceptionsParticiplesActive.js" )();
+
+var forEach = require( "lodash/forEach" );
+var includes = require( "lodash/includes" );
+var map = require( "lodash/map" );
+
+/**
+ * Creates an Participle object for the English language.
+ *
+ * @param {string} participle The participle.
+ * @param {string} sentencePart The sentence part that contains the participle.
+ * @param {object} attributes  The attributes object.
+ *
+ * @constructor
+ */
+var GermanParticiple = function(  participle, sentencePart, attributes ) {
+	Participle.call( this, participle, sentencePart, attributes );
+	this.checkException();
+};
+
+require( "util" ).inherits( GermanParticiple, Participle );
+
+/**
+ * Sets sentence part passiveness to passive if there is no exception.
+ *
+ * @returns {void}
+ */
+GermanParticiple.prototype.checkException = function() {
+	var isPassive = ! this.hasNounSuffix() && ! this.isInExceptionList() && ! this.hasHabenSeinException();
+	this.setSentencePartPassiveness( isPassive );
+};
+
+/**
+ * Checks whether a found participle is in the exception list.
+ * If a word is in the exceptionsParticiplesActive list, it isn't a participle.
+ *
+ * @returns {boolean} Returns true if it is in the exception list, otherwise returns false.
+ */
+GermanParticiple.prototype.isInExceptionList = function() {
+	var participle = this.getParticiple();
+	return includes( exceptionsParticiplesActive, participle );
+};
+
+/**
+ * Checks whether a found participle ends in a noun suffix.
+ * If a word ends in a noun suffix from the exceptionsRegex, it isn't a participle.
+ *
+ * @returns {boolean} Returns true if it ends in a noun suffix, otherwise returns false.
+ */
+GermanParticiple.prototype.hasNounSuffix = function() {
+	var participle = this.getParticiple();
+	return participle.match( exceptionsRegex ) !== null;
+};
+
+/**
+ * Checks whether a participle is followed by 'haben' or 'sein'.
+ * If a participle is followed by one of these, the sentence is not passive.
+ *
+ * @returns {boolean} Returns true if it is an exception, otherwise returns false.
+ */
+GermanParticiple.prototype.hasHabenSeinException = function() {
+	var participleIndices = getIndices( this.getParticiple(), this.getSentencePart() );
+	var habenSeinIndices = getIndicesOfList( [ "haben", "sein" ], this.getSentencePart() );
+	var isPassiveException = false;
+	if( participleIndices.length > 0 ) {
+		if ( habenSeinIndices.length === 0 ) {
+			return isPassiveException;
+		}
+		habenSeinIndices = map( habenSeinIndices, "index" );
+		forEach( participleIndices, function( participleIndex ) {
+			isPassiveException = includes( habenSeinIndices, participleIndex.index + participleIndex.match.length + 1 );
```

Comment:
I am not sure, but I don't know the size of `participleIndices`. But it might be a good idea to break this loop when `isPassiveException` became true, because that is what you want to know.